src/stan_math_backend/Transform_Mir.ml

```diff
@@ -564,7 +564,13 @@ let trans_prog (p : Program.Typed.t) =
   in
   let translate_to_open_cl stmts =
     if !use_opencl then
-      let data_var_idents = List.map ~f:fst p.input_vars in
+      let decl Stmt.Fixed.({pattern; _}) =
+      match pattern with
+        | Decl d ->
+          Some (d.decl_id)
+        | _ -> None
+      in
+      let data_var_idents = List.filter_map ~f:decl p.prepare_data in
```

Comment:
Ah, I think you'll still need the input vars, no?

Reply:
(or wait, are those included in prepare data?)

Reply:
Oh my b I assumed prepare_data had all the data, lemme check this out

Reply:
Prepare data has both! I'll add that to the opencl test